drivers/sensor/akm09918c/akm09918c.c

```diff
@@ -19,7 +20,18 @@
 
 LOG_MODULE_REGISTER(AKM09918C, CONFIG_SENSOR_LOG_LEVEL);
 
-static int akm09918c_sample_fetch(const struct device *dev, enum sensor_channel chan)
+/**
+ * @brief Perform the bus transaction to fetch samples
+ *
+ * @param dev Sensor device to operate on
+ * @param chan Channel ID to fetch
+ * @param x Location to write X channel sample or null.
+ * @param y Location to write Y channel sample or null.
+ * @param z Location to write Z channel sample or null.
+ * @return int 0 if successful or error code
+ */
+static int akm09918c_sample_fetch_helper(const struct device *dev, enum sensor_channel chan,
+					 int16_t *x, int16_t *y, int16_t *z)
```

Comment:
seems like this can be just `int16_t out[3]` instead of 3 separate pointers? You can also assert that `out` isn't `NULL` and skip some of the `if` statements below.

Reply:
This is done to accommodate the individual sample_{x,y,x} struct members in akm09918c_sample_fetch(). The if-NULL checks can probably go anyways since this is an internal function.